File: 383-ransom-note/383-ransom-note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        char[] ransomArr =  ransomNote.ToCharArray();
        char[] magArr =  magazine.ToCharArray();
        for(int i = 0; i < ransomArr.Length; i++) {
            int index = Array.IndexOf(magArr, ransomArr[i]);
            if(index != -1) {
                magArr[index] = '*';   
            }else if(index == -1) {
                return false;
            }
        }
        
        return true;
    }
}

File: 383-ransom-note/383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character, Integer> mp = new HashMap<>();
        for(int i = 0; i < magazine.length(); i++) {
            char c = magazine.charAt(i);
            if(mp.containsKey(c)) {
                mp.put(c, mp.get(c) + 1);
            }else {
                mp.put(c, 1);
            }
        }
        
        for(int i=0; i < ransomNote.length(); i++) {
            char c = ransomNote.charAt(i);
            if(mp.containsKey(c) && mp.get(c) > 0) {
                mp.put(c, mp.get(c) - 1);
            }else {
                return false;
            }
        }
        return true;
    }
}